static/demos/wgsparkl2d/index.html

```diff
@@ -3,13 +3,41 @@
 
 <body style="margin: 0px;">
     <script type="module">
-        import init from './elasticity2.js'
 
-        init().catch((error) => {
-            if (!error.message.startsWith("Using exceptions for control flow, don't mind me. This isn't actually an error!")) {
-                throw error;
+
+        function displayErrorHTML(message) {
+            const messageContainer = document.createElement("div");
+            messageContainer.style.color = "red";
+            messageContainer.style.fontWeight = "bold";
+            messageContainer.style.margin = "20px";
+            messageContainer.innerHTML = message;
+            document.body.appendChild(messageContainer);
+        }
+
+        async function check_webgpu_compatibility() {
+            if (!navigator.gpu) {
+                displayErrorHTML("WebGPU not supported. <a href='https://caniuse.com/webgpu'>More info</a>")
```

Comment:
Linking to caniuse can be controversial, alternatives would be:
- https://developer.mozilla.org/en-US/docs/Web/API/WebGPU_API#browser_compatibility is more informative but more developer-centric and "biased" towards mozilla.
- No link, leave it as an exercise for the reader

Reply:
I'll also take suggestions on wording: "This browser does not support WebGPU." might be better.

Reply:
A link to caniuse sounds fine to me.